### ISSUE TITLE
2 issues found in device management functions

### ISSUE THREAD

**Issue by kingcrimsontianyu** [ROCm-Developer-Tools/HIP#880]
For the current master branch (190e3615)
- Enum `hipFuncCache_t` and function `hipError_t hipDeviceSetCacheConfig(hipFuncCache_t cacheConfig);` in `hcc_detail/hip_runtime_api.h` are apparently defined only for HCC system, resulting in compile error on NVCC system.
- [HIP documentation](https://github.com/ROCm-Developer-Tools/HIP/blob/master/docs/markdown/CUDA_Runtime_API_functions_supported_by_HIP.md) says the equivalent of `cudaDeviceSetLimit` is supported in HIP, but apparently it is not. (The GetLimit function however is indeed provided).

**Comment by tycho** [comment]
I just ran into the first bullet point. I wrote this patch for my locally packaged HIP:

```diff
diff --git a/include/hip/nvcc_detail/hip_runtime_api.h b/include/hip/nvcc_detail/hip_runtime_api.h
index 02c4b7ee..72245399 100644
--- a/include/hip/nvcc_detail/hip_runtime_api.h
+++ b/include/hip/nvcc_detail/hip_runtime_api.h
@@ -79,6 +79,12 @@ typedef enum hipChannelFormatKind {
 #define hipBoundaryModeTrap cudaBoundaryModeTrap
 #define hipBoundaryModeClamp cudaBoundaryModeClamp
 
+// hipFuncCache
+#define hipFuncCachePreferNone cudaFuncCachePreferNone
+#define hipFuncCachePreferShared cudaFuncCachePreferShared
+#define hipFuncCachePreferL1 cudaFuncCachePreferL1
+#define hipFuncCachePreferEqual cudaFuncCachePreferEqual
+
 // hipResourceType
 #define hipResourceType cudaResourceType
 #define hipResourceTypeArray cudaResourceTypeArray
@@ -614,6 +620,10 @@ inline static hipError_t hipDeviceGetCacheConfig(hipFuncCache_t* pCacheConfig) {
     return hipCUDAErrorTohipError(cudaDeviceGetCacheConfig(pCacheConfig));
 }
 
+inline static hipError_t hipDeviceSetCacheConfig(hipFuncCache_t cacheConfig) {
+    return hipCUDAErrorTohipError(cudaDeviceSetCacheConfig(cacheConfig));
+}
+
 inline static const char* hipGetErrorString(hipError_t error) {
     return cudaGetErrorString(hipErrorToCudaError(error));
 }
```

I also noticed the documentation for `hipDeviceGetCacheConfig` in  `hcc_detail/hip_runtime_api.h` is wrong -- a copy/paste-o from  `hipFuncSetCacheConfig`.

**Comment by gargrahul** [comment]
@tycho Thanks. Could you submit your patch as a pull request?

**Comment by gargrahul** [comment]
Closing this issue since required changes were merged with #951.

**Comment by tycho** [comment]
@gargrahul that only resolved half of the issue. See 2nd bullet point in OP.
